frontend/src/metabase/browse/components/MetricsTable.tsx

```diff
@@ -0,0 +1,286 @@
+        return;
+      }
+
+      // do not trigger click when selecting text
+      const selection = document.getSelection();
+      if (selection?.type === "Range") {
+        event.stopPropagation();
+        return;
+      }
+
+      const { id, name } = metric;
+      const url = Urls.metric({ id, name });
+      const subpathSafeUrl = Urls.getSubpathSafeUrl(url);
+
+      // TODO:
+      // trackMetricClick(metric.id);
+
+      if ((event.ctrlKey || event.metaKey) && event.button === 0) {
+        Urls.openInNewTab(subpathSafeUrl);
+      } else {
+        dispatch(push(url));
+      }
+    },
+    [metric, dispatch],
+  );
+
+  return (
+    <tr onClick={handleClick} tabIndex={0}>
+      <NameCell metric={metric} />
+      <CollectionCell metric={metric} />
+      <DescriptionCell metric={metric} />
+      <Columns.RightEdge.Cell />
+    </tr>
+  );
+}
+
+function SkeletonText() {
+  return <Skeleton natural h="16.8px" />;
+}
+
+function stopPropagation(event: React.MouseEvent) {
+  event.stopPropagation();
+}
+
+function NameCell({ metric }: { metric?: MetricResult }) {
+  const headingId = `metric-${metric?.id ?? "dummy"}-heading`;
+  const icon = getIcon(metric);
+
+  return (
+    <ItemNameCell
+      data-testid="table-name"
+      aria-labelledby={headingId}
+      {...nameProps}
+    >
+      <MaybeItemLink
+        to={
+          metric ? Urls.metric({ id: metric.id, name: metric.name }) : undefined
+        }
+        style={{
+          // To align the icons with "Name" in the <th>
+          paddingInlineStart: "1.4rem",
+          paddingInlineEnd: ".5rem",
+        }}
+        onClick={stopPropagation}
+      >
+        <Icon
+          size={16}
+          {...icon}
+          color={"var(--mb-color-brand)"}
+          style={{ flexShrink: 0 }}
+        />
+        {metric ? (
+          <EntityItem.Name
+            name={metric?.name || ""}
+            variant="list"
+            id={headingId}
+          />
+        ) : (
+          <SkeletonText />
+        )}
+      </MaybeItemLink>
+    </ItemNameCell>
+  );
+}
+
+function CollectionCell({ metric }: { metric?: MetricResult }) {
+  const collectionName = metric?.collection
+    ? getCollectionName(metric.collection)
+    : t`Untitled collection`;
+
+  const content = (
+    <Flex gap="sm">
+      <FixedSizeIcon name="folder" />
+      <Box w="calc(100% - 1.5rem)">
+        {metric ? (
+          <EllipsifiedCollectionPath collection={metric.collection} />
+        ) : (
+          <SkeletonText />
+        )}
+      </Box>
+    </Flex>
+  );
+
+  return (
+    <ModelCell
+      data-testid={`path-for-collection: ${collectionName}`}
+      {...collectionProps}
+    >
+      {metric?.collection ? (
+        <Link to={Urls.collection(metric.collection)} onClick={stopPropagation}>
+          {content}
+        </Link>
+      ) : (
+        content
+      )}
+    </ModelCell>
+  );
+}
+
+function DescriptionCell({ metric }: { metric?: MetricResult }) {
```

Comment:
Should we use the same component here and in Models page?

Reply:
We will introduce more columns in a follow up PR and so the spacing and sizing of the columns might not match between the two, I can see if I can merge them when I do but would prefer not to refactor too much at this point.